packages/axios-extension/src/abap/message.ts

```diff
@@ -102,24 +102,31 @@ function logFullURL({ host, path, log }: { host: string; path?: string; log: Log
 }
 
 /**
- * Log Gateway errors returned from the S_MGW_ODATA_INNER_ERROR table which is a store of OData Inner Error data.
+ * Log Gateway errors returned from the S_MGW_ODATA_INNER_ERROR table which is a store of OData Inner Error data. In certain flows,
+ * for example, when test mode is enabled, not all error details should be displayed to the user and need to be restricted.
  *
  * @param  options options
  * @param options.error error message returned from gateway
  * @param options.log logger to be used
  * @param options.host optional host name to pretty print links
+ * @param showAllMessages optional, show all errors but restrict for certain flows i.e. test mode
  */
-export function prettyPrintError({ error, log, host }: { error: ErrorMessage; log: Logger; host?: string }): void {
+export function prettyPrintError(
+    { error, log, host }: { error: ErrorMessage; log: Logger; host?: string },
+    showAllMessages = true
```

Comment:
What is the reasoning for having it as a separate parameter instead of being part of the options object?

Reply:
To ensure it's explicit, you have change the value to disable logging. I didnt want to touch the existing optional object since this is for the normal flow of execution.